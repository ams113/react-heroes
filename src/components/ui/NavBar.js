import React, { useContext } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faUser, faSignOutAlt } from '@fortawesome/free-solid-svg-icons'

import { Link, NavLink, useHistory } from 'react-router-dom'
import { AuthContext } from '../../auth/authContext'
import { types } from '../../types/types';

export const Navbar = () => {

    const { user:{name}, dispatch } = useContext( AuthContext );
    const history = useHistory();

    const handleLogout = () => {
         
        history.replace('/login');

        dispatch({
            type: types.logout
        })
    }
    
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container">
                <Link
                    className="navbar-brand"
                    to="/"
                >
                    Publisher
                </Link>

                <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>

                <div className="collapse navbar-collapse" id="navbarSupportedContent">
                    <div className="navbar-nav mr-auto">

                        <NavLink
                            activeClassName="active"
                            className="nav-item nav-link"
                            exact
                            to="/marvel"
                        >
                            Marvel
                        </NavLink>

                        <NavLink
                            activeClassName="active"
                            className="nav-item nav-link"
                            exact
                            to="/dc"
                        >
                            DC
                        </NavLink>

                        <NavLink
                            activeClassName="active"
                            className="nav-item nav-link"
                            exact
                            to="/search"
                        >
                            Search
                        </NavLink>
                    </div>
                </div>

                <div className="navbar-collapse collapse w-100 order-3 dual-collapse2">
                    <ul className="navbar-nav ms-auto">
                        
                        <span className="nav-item nav-link text-info">
                        { name }
                      
                        </span>

                        <button
                            className="nav-item nav-link btn"
                            onClick={ handleLogout }
                        >
                            Logout
                            <i className="ms-2"><FontAwesomeIcon  icon={ faSignOutAlt } /></i>
                    </button>
                    </ul>
                </div>
            </div>
        </nav>
    )
}