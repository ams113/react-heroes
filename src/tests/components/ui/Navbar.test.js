import React from 'react';
import { mount } from 'enzyme';
import { AuthContext } from '../../../auth/authContext';
import { Navbar } from '../../../components/ui/NavBar';
import { contextLogin } from '../../fixtures/contextLogin';
import { MemoryRouter, Router } from 'react-router-dom';
import { types } from '../../../types/types';


describe('Testing Components: Navbar', () => {

    const historyMock = {
        push: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn(),
        replace: jest.fn(),
    };
    
    const wrapper = mount(
        <AuthContext.Provider value={ contextLogin }>
            <MemoryRouter>
                <Router history={ historyMock }>
                    <Navbar />
                </Router>
            </MemoryRouter>
        </AuthContext.Provider>
    );

    afterEach( ()=> {
        jest.clearAllMocks();
    });

    test('should render correctly', () => {
   
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('a').at(0).text().trim() ).toBe( 'Publisher' );
        expect( wrapper.find('a').at(1).text().trim() ).toBe( 'Marvel' );
        expect( wrapper.find('a').at(2).text().trim() ).toBe( 'DC' );
        expect( wrapper.find('a').at(3).text().trim() ).toBe( 'Search' );
        expect( wrapper.find('.text-info').text().trim() ).toBe( 'Alfred' );
    });

    test('should call logout and use history correctly', () => {
        
        // wrapper.find('button').at(1).prop('onClick')();
        wrapper.find('button').at(1).simulate('click');

        expect( contextLogin.dispatch ).toHaveBeenCalledWith({
            type: types.logout
         });

         expect( historyMock.replace ).toHaveBeenCalledWith('/login');
         
    });
    
    
})
