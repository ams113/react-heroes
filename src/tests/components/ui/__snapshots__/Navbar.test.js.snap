// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Components: Navbar should render correctly 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      onClick={[Function]}
    >
      Publisher
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarSupportedContent"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <div
        className="navbar-nav mr-auto"
      >
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ms-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Alfred
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
          <i
            className="ms-2"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-sign-out-alt fa-w-16 "
              data-icon="sign-out-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </i>
        </button>
      </ul>
    </div>
  </div>
</nav>
`;
